### ISSUE TITLE
Braid triangle mesh should not accept three mesh dims?

### ISSUE THREAD

**Issue by mclarsen** [LLNL/conduit#842]
You can ask for a explicit triangle mesh `tris` and specify the dims as `20,20,20` but the resulting mesh only has a 2d coordinate system.

```
tate: 
  time: 3.1415
  cycle: 100
coordsets: 
  coords: 
    type: "explicit"
    values: 
      x: [-10.0, -8.94736842105263, -7.89473684210526, ..., 8.94736842105263, 10.0]
      y: [-10.0, -10.0, -10.0, ..., 10.0, 10.0]
```

**Comment by mclarsen** [comment]
I assume the same thing happens for `quads`.

**Comment by cyrush** [comment]
Worked on locking this down to `npts_z == 0`, throwing an exception if folks pass anything else for a 2D only topo.

In conduit's tests -- there are several places were we pass `npts_z == 1`. 

With indirection in our tests, it's not trivial to track them all down -- but i am working on it. 

Since they were in many tests, this usage might be prevalent in the wild as well.  

Instead of an error, may use `CONDUIT_INFO` as a stop gap, switch to `CONDUIT_ERROR` in a future release

**Comment by cyrush** [comment]
resolved in #868
